examples/typescript/azure-service-bus-queue-trigger/main.ts

```diff
@@ -0,0 +1,141 @@
+  AppServicePlan,
+  AzurermProvider,
+  FunctionApp,
+  ResourceGroup,
+  ServicebusNamespace,
+  ServicebusQueue,
+  ServicebusQueueAuthorizationRule,
+  StorageAccount,
+} from "@cdktf/provider-azurerm";
+import { App, TerraformStack, Fn } from "cdktf";
+import { Construct } from "constructs";
+
+const uid = () => {
+  return "xxxxxxxxxxx".replace(/[xy]/g, function (c) {
+    var r = (Math.random() * 16) | 0,
+      v = c == "x" ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+  });
+};
+
+class MyStack extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+
+    const location = "westeurope";
+
+    // Instantiate Azure Provider
+    new AzurermProvider(this, "AzureRm", {
+      features: [{}],
+    });
+
+    // Creates the Resource Group for OCN
+    const rg = new ResourceGroup(this, "myapp-resourceGroup", {
+      name: "myapp-resourceGroup",
+      location,
+    });
+
+    // Creates the Application Insight for monitoring
+    const applicationInsight = new ApplicationInsights(
+      this,
+      "myapp-applicationInsight",
+      {
+        name: "myapp-applicationInsight",
+        location,
+        resourceGroupName: rg.name,
+        applicationType: "Node.JS",
+      }
+    );
+
+    // Creates the Service Bus Namespace
+    const namespace = new ServicebusNamespace(this, "myapp-serviceBus", {
+      name: "myapp-serviceBus",
+      location,
+      resourceGroupName: rg.name,
+      sku: "Basic",
+    });
+
+    // Creates the Service Bus Queue
+    const queue = new ServicebusQueue(this, "myapp-queue", {
+      name: "myapp-queue",
+      namespaceName: namespace.name,
+      resourceGroupName: rg.name,
+    });
+
+    // Creates the Service Bus Authorization Rule
+    const serviceBusAuthorizationRule = new ServicebusQueueAuthorizationRule(
+      this,
+      "myapp-queueAuthorizationRule",
+      {
+        name: "myapp-queueAuthorizationRule",
+        namespaceName: namespace.name,
+        resourceGroupName: rg.name,
+        queueName: queue.name,
+        listen: true,
+        send: false,
+        manage: false,
+      }
+    );
+
+    // Creates the App Service Plan
+    const azureFunctionAppServicePlan = new AppServicePlan(
+      this,
+      "myapp-appServicePlan",
+      {
+        name: "myapp-appServicePlan",
+        location,
+        resourceGroupName: rg.name,
+        sku: [
+          {
+            size: "Y1",
+            tier: "Dynamic",
+          },
+        ],
+        kind: "Linux",
+        reserved: true,
+      }
+    );
+
+    // Creates the Storage Account
+    const storageAccount = new StorageAccount(this, "myapp-storageAccount", {
+      name: `myappsa${uid()}`,
+      resourceGroupName: rg.name,
+      location,
+      accountReplicationType: "LRS",
+      accountTier: "Standard",
+    });
+
+    // Creates the Function
+    const azureFunction = new FunctionApp(this, "myapp-function", {
+      name: "myapp-function",
```

Comment:
since I don't have a lot of context in Azure and I'm curious: How is the actual code of the function deployed?

Reply:
When you deploy the function-app you deploy the "container" but not the "application code"

In fact the goal of this approach is to deploy the infrastructure and the "application code" independently as they may have different lifecycle / ownership.

The function must be deployed independently (according to the link I showed in the README.md)
I wrote an article here:
https://www.linkedin.com/pulse/embrace-gitops-cdktf-geoffrey-muselli

Do you want me to do two separate folder in this example ?

I could do:
- infrastructure: the actual code for this example
- function: a sample function and the way to deploy it